### PR TITLE
LS-21346: lightstep-tracer-go: generate uint64 trace IDs, not int64

### PR DESCRIPTION
**Pull request by neena**
R: @codeboten 
CC: @lightstep/team-ingest 

We discovered that the tracer is generating int64 trace IDs and converting them to uint64s, which means we're only using half of the trace ID space! This PR makes it so we actually generates random uint64 trace IDs. (This should halve our probability of trace ID collisions).

### CODE REVIEW

File: lightstep/rand/locked_rand.go [hunk 1]
```diff
@@ -60,6 +60,15 @@ func (lr *LockedRand) Uint64() (n uint64) {
 	return
 }
 
+// TwoUint64 generates 2 random uint64 without locking twice.
+func (lr *LockedRand) TwoUint64()  (n1, n2 uint64) {
+	lr.lk.Lock()
+	n1 = lr.r.Uint64()
+	n2 = lr.r.Uint64()
+	lr.lk.Unlock()
```
Comment (MatthewDolan) [inline lightstep/rand/locked_rand.go:68]: nit: Both here and above, I this it would be preferred to have the `Unlock()` calls be `defer`-ed.

**Review by MatthewDolan — APPROVED** [review]
LGTM.

File: lightstep/rand/locked_rand.go (hunk 1 above)
Comment (neena) [reply lightstep/rand/locked_rand.go:68]: I wanted to keep consistency with the other functions in this file!

**Comment by neena** [comment]
@codeboten did you get a chance to look at this? do you have any concerns, or can I go ahead with this?

**Review by codeboten — APPROVED** [review]
👍

**Pull request merged**